src/pal/pal_netbsd.h

```diff
@@ -24,6 +24,34 @@ namespace snmalloc
      * add new features that they should add any required feature flags.
      */
     static constexpr uint64_t pal_features = PALBSD_Aligned::pal_features;
+
+    /**
+     * Reserve memory at a specific alignment.
+     * Oddily, it appears it needs to be shared segment
+     * as it should not need to but func-memory-16 crashes otherwise.
+     */
+    template<bool committed>
+    void* reserve_aligned(size_t size) noexcept
+    {
+      // Alignment must be a power of 2.
+      SNMALLOC_ASSERT(size == bits::next_pow2(size));
+      SNMALLOC_ASSERT(size >= minimum_alloc_size);
+
+      size_t log2align = bits::next_pow2_bits(size);
+
+      void* p = mmap(
+        nullptr,
+        size,
+        PROT_READ | PROT_WRITE,
+        MAP_SHARED | MAP_ANONYMOUS | MAP_ALIGNED(log2align),
```

Comment:
Can we not do this by slightly extending the flag that is passed to mmap that comes via the template parameter in the superclass?

Reply:
Sure why not as it s most likely a temporary fix I did not bother think of it.